Specification.md

```diff
@@ -1,17 +1,145 @@
-## Gas limit
+# Specification
 
-By default `web3.js` uses `web3.eth.Contract.method.myMethod().estimateGas()`, but this value can differ from the 
-actual gas that will be used, if the contract you are calling depends on the blockhash, blocknumber or any other source of 
-randomness making the gas cost nondeterministic.
+The CLI should package several "core" extensions.
+The API these extensions expose is aimed at `node` environments.
+The aim is to provide convenience to power-users & devs.
 
-E.g.: the [`MiniMeToken`](https://github.com/aragon/aragon-apps/blob/master/shared/minime/contracts/MiniMeToken.sol) contract which snapshots balances at certain block numbers.
+## IPFS `@aragon/ipfs-utils`
 
-Considering the above-mentioned behavior, the CLI should calculate the recommended gas limit as follows:
-- `recommendedGas = estimatedGas`, if `estimatedGas > upperGasLimit` or
+Goals:
+
+- installing & starting a daemon with the right configuration
+- interacting with local/remote nodes: authentication, configuration, pinning, etc.
+- help setup a "production" node (?)
+- automatically pin anything published to an APM repository (?)
+
+### IPFS Binaries
+
+Local node:
+
+- ✔️ `aragon ipfs` - Alias for `aragon ipfs start`
+- `aragon ipfs start` - Start the **IPFS Daemon**
+  - Should start in the background and then finish
+  - Should start with the recommended configuration from `$HOMEDIR/.aragon/ipfsconfig.json`. E.g.:
+
+  ```json
+   {
+     "daemonArgs": [
+       "--migrate",
+       "--enable-namesys-pubsub"
+     ],
+     "logsLocation": "$HOMEDIR/.aragon/ipfs-logs"
+   }
+  ```
+
+  - Should save `stdout` and `stderr` to files, e.g.: `stdout-${number}.log` in the `logsLocation`
+  - Should warn if it's already running
+  - Should error (exit code 1) if it cannot start (missing libs/ports taken)
+  - Should inform about where the logs are saved and how to stop it
+  - Should run `aragon ipfs status` and finish afterwards (optional)
+- `aragon ipfs stop` - Stop the **IPFS Daemon**
+  - Should warn if the node was already stopped
+- `aragon ipfs enable-startup` - Start the **IPFS Daemon** automatically at start-up
```

Comment:
May also make sense to add the ability to clear the IPFS db. Is the plan for us to use our own DB (rather than one the user may already have from a previous IPFS installation)?

Reply:
> May also make sense to add the ability to clear the IPFS db.

👍 

> Is the plan for us to use our own DB (rather than one the user may already have from a previous IPFS installation)?

Hmm, we could totally do this, but also allow it to be the default directory. I wonder what should be the default.. 🤔